r/R/duckdb.R

```diff
@@ -124,3 +122,9 @@ duckdb_disconnector <- function(con, tbl_name) {
   })
   environment()
 }
+
+to_arrow <- function(.data) {
+  res <- DBI::dbSendQuery(dbplyr::remote_con(.data), dbplyr::remote_query(.data), arrow = TRUE)
+
+  duckdb::duckdb_fetch_record_batch(res)$read_table()
```

Comment:
We shouldn't need to read the full table here, though I haven't (yet) found a simple way to create a Dataset from a RecordBatchReader without materializing here.

Reply:
Would either of these InMemoryDataset constructors work maybe? https://github.com/apache/arrow/blob/master/cpp/src/arrow/dataset/dataset.h#L211-L216 cc @bkietz